selfapi.js

```diff
@@ -479,15 +513,24 @@ API.prototype = {
         var example = handler.examples[0];
 
         var request = example.request || {};
-        if (request.headers || request.body) {
+        if (request.headers || request.body || request.urlParameters) {
           markdown += '### Example input:\n\n';
+
+          // Format example-specific URL
+          var exampleURL = '    ' + method.toUpperCase() + ' ' + fullPath;
+          var requestParameters = request.urlParameters || {};
+          for (var requestParam in requestParameters) {
+            exampleURL = exampleURL.replace(':' + requestParam, requestParameters[requestParam]);
+          }
+          markdown += exampleURL + '\n';
+
           var requestHeaders = request.headers || {};
           for (var header in requestHeaders) {
             markdown += '    ' + header + ': ' + requestHeaders[header] + '\n';
           }
           if (request.body) {
             var requestBody = '    ' +
-              request.body.trim().replace(/\n/g, '\n    ');
+              ensureSerialized(request.body, 4).trim().replace(/\n/g, '\n    ');
```

Comment:
Note: The 4-space left-pad here is a fixed indent for Markdown. It shouldn't influence the spacing used to serialize JSON objects (which could be specified by implementors, or for example be 2 by default).